Exp 201609

feliciahalim opened the pull request:
Modify Coreutils Makefile and Add a new script to run the experiment.

feliciahalim committed:
Modify coreutils Makefile to be able executed for experiments

feliciahalim committed:
Add a script to run other Makefile, currently it run selected coreutils programs

feliciahalim committed:
Remove variable and IF conditional that is not needed in the Makefile

feliciahalim committed:
Remove IF conditional that is not needed in the Makefile

domainexpert commented on coreutils/Makefile:
```diff
@@ -118,6 +135,104 @@ sandbox:
+%.tracerx4 : 
+	KLEE_COREUTILS_MORE_OPTIONS="--search=random-path --search=nurs:covnew -max-subsumption-failure=3 -max-time=7200" OUTPUT_DIR=${CURDIR}/$@ make $*.kleetest
+
+%.kleetest: build sandbox
```
Maybe `%.klee-exp-201609` as a target name is clearer.

domainexpert commented on exp-201609.sh:
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
```
This loop can be implemented within the `coreutils/Makefile` itself, as a new `exp-201609:` target. In this way, we don't need a separate shell file.

domainexpert reviewed (changes requested):
Also, please do not forget to update `README.md` with an instruction on how to run the experiment.

feliciahalim committed:
Rename .kleetest to .klee-exp-201609 as a target name

feliciahalim committed:
Move the implementation in exp-201609.sh into coreutils/Makefile as a new exp-201609 target

feliciahalim committed:
Add sloccount in .klee-exp-201609 target

feliciahalim commented:
@domainexpert Thank you for the feedback. I have updated it accordingly with added `sloccount` in` .klee-exp-201609` target.

domainexpert commented on coreutils/Makefile:
```diff
@@ -273,8 +393,18 @@ sandbox:
 realclean: clean
 	rm -rf ${COREUTILS_GCOV_DIR} ${COREUTILS_LLVM_DIR} 
 
+exp-201609:
```
@feliciahalim Would you please add a comment here explaining this target.

domainexpert commented:
@feliciahalim Would you please also remove the WIP: from the title of this PR, as this is a PR that is complete.

feliciahalim committed:
Add comment to klee-exp-201609 target

feliciahalim committed:
Update README.md for giving instructions on how to run the exp-201609 experiment

feliciahalim commented:
@domainexpert I have updated `coreutils/Makefile ` with a comment, updated README.md by giving instructions to run exp-201609 experiment, and also updated this PR title. Thank you.

domainexpert commented:
@feliciahalim Well done! Removing the branch now ...
